File: AtlasWeb/Core/Modules/Tibbi_Surec/E_Nabiz_Modulu/KadinaYonelikSiddetVeriSetiFormViewModel.cs
//$69BAD29A
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;
using Infrastructure.Helpers;

namespace Core.Controllers
{
    public partial class KadinaYonelikSiddetVeriSetServiceController
    {
        partial void PreScript_KadinaYonelikSiddetVeriSetiForm(KadinaYonelikSiddetVeriSetiFormViewModel viewModel, KadinaYonelikSiddetVeriSet kadinaYonelikSiddetVeriSet, TTObjectContext objectContext)
        {
            Guid? selectedEpisodeActionID = viewModel.GetSelectedEpisodeActionID();
            if (selectedEpisodeActionID.HasValue)
            {
                EpisodeAction episodeAction = objectContext.GetObject<EpisodeAction>(selectedEpisodeActionID.Value);
                viewModel._KadinaYonelikSiddetVeriSet.EpisodeAction = episodeAction;
                viewModel._KadinaYonelikSiddetVeriSet.SubEpisode = episodeAction.SubEpisode;

            }
            ContextToViewModel(viewModel, objectContext);
        }
    }
}

namespace Core.Models
{
    public partial class KadinaYonelikSiddetVeriSetiFormViewModel: BaseViewModel, IENabizViewModel
    {
        public void AddENabizObjectViewModelToContext(TTObjectContext objectContext)
        {
            objectContext.AddObject(this._KadinaYonelikSiddetVeriSet);

            if (this.KadinaYonelikSiddetSonucGridList != null)   // Ek nesneler için
            {
                foreach (var item in this.KadinaYonelikSiddetSonucGridList)
                {
                    var kadinaYonelikSiddetSonuc = (KadinaYonelikSiddetSonuc)objectContext.AddObject(item);
                    kadinaYonelikSiddetSonuc.KadinaYonelikSiddetVeriSet = this._KadinaYonelikSiddetVeriSet;
                }
            }

            if (this.KadinaYonelikSiddetTuruGridList != null)   // Ek nesneler için
            {
                foreach (var item in this.KadinaYonelikSiddetTuruGridList)
                {
                    var kadinaYonelikSiddetTuru = (KadinaYonelikSiddetTuru)objectContext.AddObject(item);
                    kadinaYonelikSiddetTuru.KadinaYonelikSiddetVeriSet = this._KadinaYonelikSiddetVeriSet;
                }
            }

           


            var kadinaYonelikSiddetVeriSet = (KadinaYonelikSiddetVeriSet)objectContext.GetLoadedObject(this._KadinaYonelikSiddetVeriSet.ObjectID);


            new SendToENabiz(objectContext, kadinaYonelikSiddetVeriSet.SubEpisode, kadinaYonelikSiddetVeriSet.ObjectID, kadinaYonelikSiddetVeriSet.ObjectDef.Name, "231", Common.RecTime());
        }
    }
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/E_Nabiz_Modulu/KadinaYonelikSiddetVeriSetiFormViewModel.ts
//$BDA46EB9
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { KadinaYonelikSiddetVeriSet } from "NebulaClient/Model/AtlasClientModel";
import { KadinaYonelikSiddetTuru } from "NebulaClient/Model/AtlasClientModel";
import { KadinaYonelikSiddetSonuc } from "NebulaClient/Model/AtlasClientModel";
import { SKRSSiddetTuru } from "NebulaClient/Model/AtlasClientModel";
import { SKRSKadinaYonelikSiddetSonucuYonlendirmeveDegerlendirme } from "NebulaClient/Model/AtlasClientModel";
import { Type } from 'NebulaClient/ClassTransformer';

export class KadinaYonelikSiddetVeriSetiFormViewModel extends BaseViewModel {
    protected __type__: string = 'Core.Models.KadinaYonelikSiddetVeriSetiFormViewModel, Core';
    @Type(() => KadinaYonelikSiddetVeriSet)
    public _KadinaYonelikSiddetVeriSet: KadinaYonelikSiddetVeriSet = new KadinaYonelikSiddetVeriSet();
    @Type(() => KadinaYonelikSiddetTuru)
    public KadinaYonelikSiddetTuruGridList: Array<KadinaYonelikSiddetTuru> = new Array<KadinaYonelikSiddetTuru>();
    @Type(() => KadinaYonelikSiddetSonuc)
    public KadinaYonelikSiddetSonucGridList: Array<KadinaYonelikSiddetSonuc> = new Array<KadinaYonelikSiddetSonuc>();
    @Type(() => SKRSSiddetTuru)
    public SKRSSiddetTurus: Array<SKRSSiddetTuru> = new Array<SKRSSiddetTuru>();
    @Type(() => SKRSKadinaYonelikSiddetSonucuYonlendirmeveDegerlendirme)
    public SKRSKadinaYonelikSiddetSonucuYonlendirmeveDegerlendirmes: Array<SKRSKadinaYonelikSiddetSonucuYonlendirmeveDegerlendirme> = new Array<SKRSKadinaYonelikSiddetSonucuYonlendirmeveDegerlendirme>();
}
